src/main/java/com/imsweb/layout/naaccrxml/NaaccrXmlLayout.java

```diff
@@ -0,0 +1,467 @@
+        if (data == null)
+            return;
+
+        PatientXmlWriter writer = null;
+        try {
+            writer = new PatientXmlWriter(new OutputStreamWriter(outputStream, StandardCharsets.UTF_8), data);
+            writeAllPatients(writer, patients);
+        }
+        finally {
+            if (writer != null)
+                writer.closeAndKeepAlive();
+        }
+    }
+
+    /**
+     * Used to write a list of patients using a PatientXmlWriter. Can be multiple times to write to the same file.
+     * This method does not open or close the writer.
+     * @param writer The PatientXmlWriter used to write Patients to an XML file
+     * @param patients List of Patients to be written
+     * @throws NaaccrIOException
+     */
+    public void writeAllPatients(PatientXmlWriter writer, List<Patient> patients) throws NaaccrIOException {
+        if (patients != null)
+            for (Patient patient : patients)
+                writer.writePatient(patient);
+    }
+
+    /**
+     * Used to read single patients from an XML file using a PatientXmlReader
+     * @param reader PatientXmlReader used to read Patients from an XML file.
+     * @return returns the next Patient found by the reader
+     * @throws NaaccrIOException
+     */
+    public Patient readNextPatient(PatientXmlReader reader) throws NaaccrIOException {
+        return reader.readPatient();
+    }
+
+    /**
+     * Opens a FileInputStream to read all the Patients from an XML file. Passes the stream to the method below.
+     * The method opens and closes the InputStream.
+     * @param file XML file that Patients will be read from.
+     * @param encoding The encoding of the file (null means default OS encoding)
+     * @return List of all Patients in the file.
+     * @throws IOException
+     */
+    public List<Patient> readAllPatients(File file, String encoding) throws IOException {
+        List<Patient> patients;
+
+        try (FileInputStream is = new FileInputStream(file)) {
+            patients = readAllPatients(is, encoding);
+        }
+        return patients;
+    }
+
+    /**
+     * Creates a PatientXmlReader from the InputStream parameter to read all Patients from an XML file. Passes the writer to the method below.
+     * This method opens and closes the reader.
+     * This method does not open or close the InputStream.
+     * @param inputStream InputStream used to read Patients
+     * @param encoding Stream encoding (null means default OS encoding)
+     * @return List of all Patients read from the stream.
+     * @throws NaaccrIOException
+     * @throws UnsupportedEncodingException
+     */
+    public List<Patient> readAllPatients(InputStream inputStream, String encoding) throws NaaccrIOException, UnsupportedEncodingException {
+        NaaccrOptions options = new NaaccrOptions();
+        options.setUseStrictNamespaces(false);
```

Comment:
Did you do this because of your unit test? I don't think you should assume this. The NaaccrOptions object has defaults that are supposed to make sense. We should use those defaults, but also allow a user to change them. Please add a NaaccrOptions as a parameter to every read/write method that creates a reader/writer. The NAACCR XML library should already handle a null options object, so you shouldn't need to worry about that. If you want the unit tests to not care about strict namespaces, then you should set that options in the tests.